### PR TITLE
ignore foreign aliases

### PR DESCRIPTION
**Pull request by laerus**
that match xonsh bultin aliases

#1612 #1155

### CODE REVIEW

File: xonsh/foreign_shells.py [hunk 1]
```diff
@@ -540,9 +541,20 @@ def load_foreign_aliases(shells=None, config=None, issue_warning=True):
...
-            aliases.update(shaliases)
+        if not shaliases:
+            continue
+        for alias in list(shaliases.keys()):
+            if alias not in xonsh_aliases:
+                continue
+            else:
+                del shaliases[alias]
+                print('aliases: alias {!r} of shell {!r} '
+                      'tries to override xonsh builtin alias, '
```
Comment (scopatz) [inline xonsh/foreign_shells.py]: These aren't only just builtin aliases.  They are just any pre-existing aliases.

File: xonsh/foreign_shells.py [hunk 2]
```diff
@@ -540,9 +541,20 @@ def load_foreign_aliases(shells=None, config=None, issue_warning=True):
...
+                      'xonsh wins!'.format(alias, shell['shell']),
+                      file=sys.stderr)
```
Comment (scopatz) [inline xonsh/foreign_shells.py]: Also, couldn't this be:

```python
for alias in set(shaliases) & set(xonsh_aliases):
    del shaliases[alias]
    print(....)
```

It seems like this is a set operation, so maybe should be expressed as one.  Just a thought!

**Comment by scopatz** [comment]
Thanks a ton @laerus! Just a couple of comments in-line!

File: xonsh/foreign_shells.py (hunk 1 above)
Comment (laerus) [inline xonsh/foreign_shells.py]: nice!

File: xonsh/foreign_shells.py (hunk 2 above)
Comment (laerus) [inline xonsh/foreign_shells.py]: right!

File: xonsh/foreign_shells.py [hunk 3]
```diff
@@ -540,9 +541,17 @@ def load_foreign_aliases(shells=None, config=None, issue_warning=True):
...
     shells = _get_shells(shells=shells, config=config, issue_warning=issue_warning)
     aliases = {}
+    xonsh_aliases = builtins.aliases
     for shell in shells:
         shell = ensure_shell(shell)
         _, shaliases = foreign_shell_data(**shell)
-        if shaliases:
-            aliases.update(shaliases)
+        if not shaliases:
+            continue
```
Comment (scopatz) [inline xonsh/foreign_shells.py]: You don't need these the above two lines anymore either because `set(shaliases) == set()` and so the loop will be empty.

Comment (scopatz) [inline xonsh/foreign_shells.py]: This doesn't block the PR, just an opportunity to get rid of a couple more lines :)

Comment (laerus) [inline xonsh/foreign_shells.py]: Always good to remove useless lines!

**Comment by scopatz** [comment]
Amazing! Thanks so much!

**Comment by laerus** [comment]
thanks for the intersection hint!

**Pull request merged**